contracts/V1/StateChannelDiamondProxy/DisputeManagerFacet.sol

```diff
@@ -99,121 +99,168 @@ contract DisputeManagerFacet is StateChannelCommon {
                 .executeStateTransitionOnState(channelId, encodedState, _tx);
     }
 
-    function getDispute(
-        bytes32 channelId
-    ) public view returns (Dispute memory) {
-        return disputes[channelId];
+    
+
+    function _handleBlockFraudProofs(
+       
+    ) internal {
+      
+       
+
+    }
+
+    function _handleDisputeFraudProofs(
+        // core specific checks
+        // apply proofs
+        // apply dispute to latest state
+        // set state
+    ) internal {
+
+    }
+
+    function _handleTimeoutDispute(
+
+    ) internal {
+
     }
 
     function createDispute(
         bytes32 channelId,
         uint forkCnt,
-        bytes memory encodedLatestFinalizedState,
-        bytes memory encodedLatestCorrectState,
-        ConfirmedBlock[] memory virtualVotingBlocks,
-        address timedoutParticipant,
-        uint foldedTransactionCnt,
-        Proof[] memory proofs
-    ) public {
-        require(!isDisputeInProgress(channelId), ErrorDisputeInProgrees());
-        require(getForkCnt(channelId) == forkCnt, ErrorDisputeForkMismatch());
+        bytes calldata encodedLatestState,
+        StateProof[] calldata stateProofs,
+        address[] calldata onChainSlashes,
```

Comment:
This should literally be the Dispute Struct from the docs and it should check msg.sender == dispute.disputer. 

Other way to do it is omit/remove the 'challengeDeadline' from the struct and submit a signedDispute (which would be better) since we want to collect signatures to hopefully finalize early. The timestamp/deadline can be used the same way as with blockCalldata when calculating the commitment (hash(dispute,block.timestamp)).

Reply:
second argument to the function call (dispute is first) should be 'shouldInvokeAuditing' that defaults to false.